test/e2e/utils.go

```diff
@@ -261,6 +262,26 @@ func currentState(namespace string, node string, currentStateYaml *nmstatev1alph
 	}, ReadTimeout, ReadInterval)
 }
 
+// TODO log actual condition when fails
+func checkCondition(node string, conditionType nmstatev1alpha1.NodeNetworkStateConditionType, conditionStatus corev1.ConditionStatus) {
+	key := types.NamespacedName{Name: node}
+	var condition *nmstatev1alpha1.NodeNetworkStateCondition
+	Eventually(func() *nmstatev1alpha1.NodeNetworkStateCondition {
```

Comment:
Could you merge following two Eventually? You can get the state, check for the condition, if it is nil return ConditionStatus as corev1.ConditionUnknown (I think it should be in corev1).